server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -90,44 +88,25 @@ public class KillUnusedSegments implements CoordinatorDuty
   private final OverlordClient overlordClient;
 
   public KillUnusedSegments(
-      final SegmentsMetadataManager segmentsMetadataManager,
-      final OverlordClient overlordClient,
-      final DruidCoordinatorConfig config
+      SegmentsMetadataManager segmentsMetadataManager,
+      OverlordClient overlordClient,
+      KillUnusedSegmentsConfig killConfig
   )
   {
-    if (config.getCoordinatorKillPeriod().getMillis() < config.getCoordinatorIndexingPeriod().getMillis()) {
-      throw DruidException.forPersona(DruidException.Persona.OPERATOR)
-                          .ofCategory(DruidException.Category.INVALID_INPUT)
-                          .build(
-                              StringUtils.format(
-                                     "druid.coordinator.kill.period[%s] is invalid. It must be greater than or "
-                                     + "equal to druid.coordinator.period.indexingPeriod[%s].",
-                                     config.getCoordinatorKillPeriod(),
-                                     config.getCoordinatorIndexingPeriod()
-                                 )
-                          );
-    }
-    if (config.getCoordinatorKillMaxSegments() < 0) {
-      throw DruidException.forPersona(DruidException.Persona.OPERATOR)
-                          .ofCategory(DruidException.Category.INVALID_INPUT)
-                          .build(StringUtils.format(
-                                     "druid.coordinator.kill.maxSegments[%d] is invalid. It must be a positive integer.",
-                                     config.getCoordinatorKillMaxSegments()
-                                 )
-                          );
-    }
-    this.period = config.getCoordinatorKillPeriod();
-    this.ignoreDurationToRetain = config.getCoordinatorKillIgnoreDurationToRetain();
-    this.durationToRetain = config.getCoordinatorKillDurationToRetain();
+    this.period = killConfig.getCleanupPeriod();
+
+    this.maxSegmentsToKill = killConfig.getMaxSegments();
+    this.ignoreDurationToRetain = killConfig.isIgnoreDurationToRetain();
+    this.durationToRetain = killConfig.getDurationToRetain();
     if (this.ignoreDurationToRetain) {
       log.info(
           "druid.coordinator.kill.durationToRetain[%s] will be ignored when discovering segments to kill "
           + "because druid.coordinator.kill.ignoreDurationToRetain is set to true.",
-          this.durationToRetain
+          durationToRetain
       );
     }
-    this.bufferPeriod = config.getCoordinatorKillBufferPeriod().getMillis();
-    this.maxSegmentsToKill = config.getCoordinatorKillMaxSegments();
+    this.bufferPeriod = killConfig.getBufferPeriod().getMillis();
```

Comment:
nit: taking a fresh look, it looks like we can store the buffer period directly as-is without the millis part (or rename it to `bufferPeriodMillis` to avoid ambiguity)

Reply:
Done.